tasks/upload.py

```diff
@@ -381,54 +379,40 @@ def run_impl(
             retry_countdown = 20 * 2**self.request.retries
             log.warning(
                 "Retrying upload",
-                extra=dict(
-                    commit=commitid,
-                    repoid=repoid,
-                    report_type=report_type,
-                    countdown=int(retry_countdown),
-                ),
+                extra=upload_context.log_extra(countdown=int(retry_countdown)),
+            )
+            self.retry(
+                max_retries=3,
+                countdown=retry_countdown,
+                kwargs=upload_context.kwargs_for_retry(kwargs),
             )
-            upload_context.prepare_kwargs_for_retry(kwargs)
-            self.retry(max_retries=3, countdown=retry_countdown, kwargs=kwargs)
 
     @sentry_sdk.trace
     def run_impl_within_lock(
         self,
-        db_session,
+        db_session: Session,
         upload_context: UploadContext,
-        *args,
-        **kwargs,
+        checkpoints: CheckpointLogger | None,
+        kwargs: dict,
     ):
-        log.info(
-            "Starting processing of report",
-            extra=dict(
-                repoid=upload_context.repoid,
-                commit=upload_context.commitid,
-                report_type=upload_context.report_type.value,
-                report_code=upload_context.report_code,
-            ),
-        )
+        log.info("Starting processing of report", extra=upload_context.log_extra())
         repoid = upload_context.repoid
-        commitid = upload_context.commitid
         report_type = upload_context.report_type
-        report_code = upload_context.report_code
 
-        checkpoints = None
         if report_type == ReportType.COVERAGE:
             try:
-                checkpoints = checkpoints_from_kwargs(UploadFlow, kwargs)
+                assert checkpoints
```

Comment:
I think this raises `AssertionError` instead of `ValueError`, right?...

Reply:
to be quite honest, I don’t know what this would raise, because this never should. if it ever does, it is a programming error that should be fixed immediately.